### DISCUSSION TITLE
New to keygen; What do Entitlements do?

### DISCUSSION THREAD

**Discussion by maaz-yobe** [keygen-sh/community#134] [Q&A]
Hi everyone!

I've recently started using keygen and am exploring how to implement it. One thing that confuses me is what are entitlements and what are they used for? The policy seems to contain all the rules that can be associated with different licenses which in turn can be assigned a user, but then entitlements can be assigned to both the license and policy. What does it do for the licensing process? Is it necessary?

**Comment by ezekg** [comment]
Entitlements, in their simplest form, are feature flags. You can attach entitlements to policies to propagate the policy's entitlements to all licenses implementing that policy. You can attach entitlements to licenses for extra one-off entitlements per-license.

They aren't required. If you need more complex feature flags than a set of key-values in a license's `metadata`, you can reach for entitlements.
